test/Microsoft.AspNet.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs

```diff
@@ -214,5 +214,67 @@ private class Address2
             var error = Assert.Single(modelState[street].Errors);
             Assert.Equal("The Street field is required.", error.ErrorMessage);
         }
+
+        private class Person3
+        {
+            [FromBody]
+            public Address3 Address { get; set; }
+        }
+
+        private class Address3
+        {
+            public string Street { get; set; }
+
+            [Required]
+            public int Zip { get; set; }
+        }
+
+        [Theory(Skip = "There should be entries for all model properties which are bound. #2445")]
+        [InlineData("{ \"Street\" : \"someStreet\" }")]
+        [InlineData("{}")]
+        public async Task FromBodyOnProperty_RequiredOnValueTypeSubProperty_AddsModelStateError(string inputText)
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                BindingInfo = new BindingInfo()
+                {
+                    BinderModelName = "CustomParameter",
+                },
+                ParameterType = typeof(Person3)
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(
+                request =>
+                {
+                    request.Body = new MemoryStream(Encoding.UTF8.GetBytes(inputText));
+                    request.ContentType = "application/json";
+                });
+            var httpContext = operationContext.HttpContext;
+            var actionContext = httpContext.RequestServices.GetRequiredService<IScopedInstance<ActionContext>>().Value;
+            var modelState = actionContext.ModelState;
```

Comment:
This fix was checked in, you don't need to use `IScopedInstance<ActionContext>` here anymore